diff --git a/elfio/elf_types.hpp b/elfio/elf_types.hpp
index ff6743f..72ff015 100644
--- a/elfio/elf_types.hpp
+++ b/elfio/elf_types.hpp
@@ -1165,7 +1165,9 @@ constexpr Elf_Word AT_RANDOM        = 25; // Address of 16 random bytes.
 constexpr Elf_Word AT_HWCAP2  = 26; // More hints about processor capabilities.
 constexpr Elf_Word AT_EXECFN  = 31; // Filename of executable.
 constexpr Elf_Word AT_SYSINFO = 32; // EP to the system call in the vDSO.
-constexpr Elf_Word AT_SYSINFO_EHDR = 33; // Start of the ELF header of the vDSO.
+#ifndef AT_SYSINFO_EHDR
+constexpr Elf_Word AT_SYSINFO_EHDR = 33;  // Start of the ELF header of the vDSO.
+#endif
 constexpr Elf_Word AT_L1I_CACHESHAPE    = 34;
 constexpr Elf_Word AT_L1D_CACHESHAPE    = 35;
 constexpr Elf_Word AT_L2_CACHESHAPE     = 36;
